cli/LineReaderBuffered.cpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ **/
+
+#include "cli/LineReaderBuffered.hpp"
+#include <iostream>
+
+namespace quickstep {
+
+LineReaderBuffered::LineReaderBuffered(const std::string &default_prompt,
+                                       const std::string &continue_prompt)
+  : LineReader("", ""),
+    buffer_empty_(true) { }
+
+LineReaderBuffered::LineReaderBuffered() : LineReader("", ""), buffer_empty_(true) { }
+
+std::string LineReaderBuffered::getLineInternal(const bool continuing) {
+  // This method is called when the leftover_ string is depleted.
+  buffer_empty_ = true;
+  return "";
+}
+
+}
+
+
+/*
+TEST(NetworkIOTest, TestLineReaderBuffered) {
+  // The buffered line reader is used exclusively by the NetworkIO.
+  LineReaderBuffered linereader;
+  EXPECT_TRUE(linereader.bufferEmpty());
+
+  std::string stmt_1 = "select * from foo;";
+  std::string stmts = stmt_1 + "select 1; select 2; quit;";
+  linereader.setBuffer(stmts);
+  ASSERT_FALSE(linereader.bufferEmpty());
+
+  std::vector<std::string> stmts_vec;
+  int parsed_commands;
+  for(parsed_commands = 0;
+      parsed_commands < 10 && !linereader.bufferEmpty();
+      parsed_commands++) {
+    std::string command = linereader.getNextCommand();
+    if (command != "") {
+      stmts_vec.push_back(command);
+    }
+  }
+
+  EXPECT_EQ(stmt_1, stmts_vec.front());
+  EXPECT_EQ(4, stmts_vec.size());
+  EXPECT_TRUE(linereader.bufferEmpty());
+}
+*/
```

Comment:
..well, actually I moved the commented out comments back into the unit test. Originally I had written this LRBuffered class, then removed it, placing the files somewhere safe. When I moved them back in, I forgot that I stashed the test in the file.